testsuite/objects/__init__.py

```diff
@@ -99,6 +99,33 @@ class Rule:
     value: str
 
 
+@dataclass
+class AnyPatterns:
+    """Dataclass specifying *OR* operation on patterns. Any one needs to pass for this block to pass."""
+
+    any: list["Patterns"]
+
+
+@dataclass
+class AllPatterns:
+    """Dataclass specifying *AND* operation on patterns. All need to pass for this block to pass."""
+
+    all: list["Patterns"]
+
+
+@dataclass
+class PatternRef:
+    """
+    Dataclass that references other pattern-matching expression by name.
+    Use authorization.add_patterns() function to define named pattern-matching expression.
+    """
+
+    patternRef: str
+
+
+Patterns = Pattern | AnyPatterns | AllPatterns | PatternRef
```

Comment:
Not sure how it is called in the AC/AP directly, but if we would like to preserve the name, we could just name this alias a `Rule`, which would limit changes needed for this PR

Reply:
Renamed the type to `Rule`